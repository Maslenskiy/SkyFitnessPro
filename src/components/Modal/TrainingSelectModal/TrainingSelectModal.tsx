import React, { useEffect, useState } from "react";
import "./TrainingSelectModal.css";
import TrainingLink from "./TrainingLink/TrainingLink";
import { getCourseById, getWorkoutsById } from "../../../utils/api";
import { Exercise } from "../../../types/training";

interface ModalProps {
	closeModal: () => void;
	courseId: string;
}

const TrainingSelectModal: React.FC<ModalProps> = ({ closeModal, courseId }) => {
	const [courseData, setCourseData] = useState([]);
	const [workoutInfo, setWorkoutInfo] = useState<Exercise[]>([]);
	const [isLoaded, setIsLoaded] = useState(false);

	useEffect(() => {
		getCourseById(courseId)
			.then((data) => {
				setCourseData(data.workouts);
			})
			.catch((error: unknown) => console.error(error));
	}, [courseId]);

	useEffect(() => {
		async function fetchWorkoutInfo() {
			if (courseData.length > 0) {
				try {
					const workoutInfoArray = await Promise.all(
						courseData.map(async (workout) => {
							const response = await getWorkoutsById(workout);
							return response;
						}),
					);
					setWorkoutInfo(workoutInfoArray);
					setIsLoaded(true);
				} catch (error: unknown) {
					console.error("Ошибка при получении информации о курсе:", error);
				}
			}
		}
		fetchWorkoutInfo();
	}, [courseData]);

	return (
		<div className="fixed z-40 inset-0 flex justify-center items-center bg-gray-900 bg-opacity-50" onClick={closeModal}>
			{isLoaded ? (
				<div
					className="relative flex flex-col items-center p-8 gap-8 w-[355px] bg-white shadow-[0_4px_67px_-12px_rgba(0,0,0,0.13)] rounded-[30px] overflow-hidden"
					onClick={(e) => e.stopPropagation()}
				>
					<div className="flex flex-col items-start gap-3 w-full h-full">
						<h2 className="text-black text-[34px] text-start leading-10 font-medium mb-[20px]">Выберите тренировку</h2>

						<div className="flex flex-col gap-3 w-full h-[374px] overflow-y-auto leading-5 pr-[20px]">
							{workoutInfo.map((workout, index) => (
								<TrainingLink key={index} trainingId={workout._id} name={workout.name} courseId={courseId} />
							))}
						</div>
					</div>
				</div>
			) : (
				<p className="text-black text">Загрузка</p>
			)}
		</div>
	);
};

export default TrainingSelectModal;
