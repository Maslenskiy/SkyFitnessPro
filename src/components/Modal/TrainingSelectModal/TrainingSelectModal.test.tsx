import { render } from "@testing-library/react";
import TrainingSelectModal from "./TrainingSelectModal";
import "@testing-library/jest-dom/";

// Мокируем функции для API
jest.mock("../../../utils/api", () => ({
	getCourseById: jest.fn().mockResolvedValue({ workouts: ["workout1", "workout2"] }),
	getWorkoutsById: jest.fn().mockResolvedValue({ _id: "workout1", name: "Workout 1" }),
}));

describe("TrainingSelectModal", () => {
	it("matches the snapshot", () => {
		const { asFragment } = render(<TrainingSelectModal closeModal={() => {}} courseId="course1" />);

		expect(asFragment()).toMatchSnapshot();
	});
});
