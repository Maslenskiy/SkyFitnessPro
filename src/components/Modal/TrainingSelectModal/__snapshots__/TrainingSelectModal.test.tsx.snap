// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrainingSelectModal matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="fixed z-40 inset-0 flex justify-center items-center bg-gray-900 bg-opacity-50"
  >
    <p
      class="text-black text"
    >
      Загрузка
    </p>
  </div>
</DocumentFragment>
`;
